### ISSUE TITLE
Error: Function::getValue(),getDerivative,getExpr - invalid state

### ISSUE THREAD

**Issue by hrani** [BhallaLab/moose-core#409]
With the latest **fix/feature** for function, now when created a new instance of function, getting error 
as 

> Error: Function::getDerivative() - invalid state

> Error: /m[0]/c[0]/f[0]::getExpr() - invalid parser state

> Expression was : 

> Error: Function::getValue() - invalid state

> warn: Expr is empty 

`
[functionerror.txt](https://github.com/BhallaLab/moose-core/files/4475693/functionerror.txt)

`

**Comment by dilawar** [comment]
Fix are pending in #405 .

**Comment by dilawar** [comment]
Duplicate of #401

**Comment by dilawar** [comment]
For this particular example, you can set `expr` to `0`. Empty expression is a warning.

```
moose.Neutral('/m')
moose.CubeMesh('/m/c')
f = moose.Function('/m/c/f')
f.expr = '0'  # don't leave it blank.
moose.showfield(f)
```

**Comment by hrani** [comment]
When one created a new function object one can't expect to fill the values before hand.
Default values need to be filled.

**Comment by dilawar** [comment]
Ok. The old behaviour in 3.1.5 is that if expression is not set it is 0. I'll update #405 to change this behaviour.

```python
>>> import moose
f = moo>>> 
[dilawars@localhost ~]$ PYTHONPATH='' python3
Python 3.8.2 (default, Mar 05 2020, 18:58:42) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import moose
>>> f = moose.Function('f1')
>>> moose.showfields(f)

[ /f[0] ]
className        = Function
derivative       = 0.0
dt               = 0.1
expr             = 0 
fieldIndex       = 0
idValue          = 452
independent      = x0
index            = 0
mode             = 1
name             = f
numData          = 1
numField         = 1
numVars          = 0
path             = /f[0]
rate             = 0.0
tick             = 12
value            = 0.0
>>>
```

**Comment by dilawar** [comment]
Fix are pending in #405 . Once travis passes it, please review and merge it.
